src/main/java/seedu/cafectrl/parser/Parser.java

```diff
@@ -72,11 +72,12 @@ public class Parser implements ParserUtil {
             + "qty/([A-Za-z0-9\\s]+)";
 
     /** The rest of Command Handler Patterns*/
+
     private static final String LIST_INGREDIENTS_ARGUMENT_STRING = "dish/(.+)";
-    private static final String DELETE_ARGUMENT_STRING = "(\\d+)";
+    private static final String DELETE_ARGUMENT_STRING = "(.+)";
     private static final String EDIT_PRICE_ARGUMENT_STRING = "dish/(.*)\\sprice/(.*)";
-    private static final String BUY_INGREDIENT_ARGUMENT_STRING = "(ingredient/[A-Za-z0-9\\s]+ qty/[A-Za-z0-9\\s]+"
-            + "(?:, ingredient/[A-Za-z0-9\\s]+ qty/[A-Za-z0-9\\s]+)*)";
+    private static final String BUY_INGREDIENT_ARGUMENT_STRING = "(ingredient/[A-Za-z0-9\\\\s]+ qty/.+"
+            + "(?:, ingredient/[A-Za-z0-9\\\\s]+ qty/.+)*)";
```

Comment:
perhaps you should remove the extra '\\' in the regex
```suggestion
    private static final String BUY_INGREDIENT_ARGUMENT_STRING = "(ingredient/[A-Za-z0-9\\s]+ qty/.+"
            + "(?:, ingredient/[A-Za-z0-9\\s]+ qty/.+)*)";
```